### DISCUSSION TITLE
More granular control over activity and social features hidden with private profiles

### DISCUSSION THREAD

**Discussion by ashn-dot-dev** [community/community#15365] [Other Feature Feedback, Questions, & Ideas]
The new [private profile](https://github.blog/changelog/2022-04-21-private-profiles/) feature currently in beta is a welcomed edition to the GitHub platform. However, setting a profile to private is currently an all-or-nothing option. As a user, I would like to individually choose which activity and social features are made public and which are kept private on a feature-by-feature basis.

**Concrete Example:**

I use GitHub as both a Git hosting platform and as a supplement to my résumé. I would like other users to be able to discover my software through the trending page, and I would like potential employers to be able to see selected projects through pinned repositories on my profile. But I would also like hide my activity overview/feed, contribution graph, follower/following activity, and starred projects, since that information is more personal and not something I am completely comfortable with having so prominently displayed. Currently, the private profile feature does not allow this combination (or similar) of data to be selectively displayed.

**Comment by sameersismail** [comment]
Agreed—more granular controls would be great.

> a supplement to my résumé [...]  I would like potential employers to be able to see selected projects through pinned repositories on my profile

This could be emulated coarsely via the `github.com/$USER/$USER` repository.

**Comment by crystalcommunication** [comment]
I agree completely. I was very pleased to learn that they had finally added an option to hide your contribution history and other profile information. What I do not understand is why the option also hides my pinned repositories. I have no problem with this being an option, but generally, if I didn't want people to see my pinned repositories, I wouldn't have pinned them. It makes sense to hide this if it uses the same container as the popular repositories section, which people might want to hide, but it should be optional.

As mentioned above, the private profile does not hide your profile's README if you have created one. This is really good, you can just drop a list of links to notable repositories in there, but the user experience could only stand to improve by having some additional granularity to what information is publicly available on your profile.

In any case, what we now have is miles better than all profiles being public and having all public contribution graphs. Big thanks to GitHub for finally implementing this long overdue feature.

**Comment by easa** [comment]
Personally, I don't like private profile!
GitHub is not a social media; private option add no value to it. 
Instead I like stats section (LoC per day, day of week with no contribution, LoC in languages, issue per day, etc ) so I could see my productivity. Maybe team could use the stats section to assign stories and more...

**Reply by crystalcommunication** [reply]
I strongly disagree. GitHub does behave like a social media platform in many ways, and the gamification of public contributions on your profile is an element of that. Moreover, other code collaboration platforms do have private profile options, and many of us have enjoyed having our profiles private since it was implemented. I strongly reject the reasoning that the option has no value. Perhaps most importantly is the fact that it is an **option**, meaning that you can disable it if you don't like it for your profile, but for those of us who don't care to have people snooping around our public contributions on our profiles, it is a valuable **option**.

If you want more information and statistics added to the profile data, that seems like a fine idea, but it has absolutely nothing to do with privacy controls.

**Reply by itsjavi** [reply]
> GitHub is not a social media

precisely because of that, we need privacy options.

**Comment by 4015-al** [comment]
I add to this, that I'd like to be able to keep some Public Profile info publicly available, while using a private profiles. (eg: my public email address)

**Comment by itsjavi** [comment]
I completely agree with the proposal, and I want to add that, IMHO, repo Star ⭐️ lists should have it's own public/private visibility (the same as project boards).

**Comment by CaseyLabs** [comment]
Adding my +1 here as I just encountered this issue.

**Comment by vojkny** [comment]
Related issue: https://github.com/orgs/community/discussions/31372 - would be great if both could be handled.
